src/Razor/src/Microsoft.AspNetCore.Razor.Language/Syntax/SyntaxNodeExtensions.cs

```diff
@@ -243,15 +243,33 @@ private static string GetTagNameCore(RazorBlockSyntax tagBlock)
                 throw new ArgumentNullException(nameof(tagBlock));
             }
 
-            if (tagBlock.Children.Count == 0 || !(tagBlock.Children[0] is MarkupTextLiteralSyntax childLiteral))
+            MarkupTextLiteralSyntax nameLiteral = null;
```

Comment:
I know all this logic will change when I split `MarkupStartTag` and `MarkupEndTag`. But I want to do that change last so as to not complicate things.

Reply:
Sure.